### ISSUE TITLE
Follow Users Who Liked Content Of User

### ISSUE THREAD

**Issue by RFlamant** [mifi/instauto#85]
Based on #75 may you provide a way to list content_shortcodes from a user, so we get an input for the function getUsersWhoLikedContent({ contentId })

Awaited workflow for FollowUsersWhoLikedContentOfUser : 
- 1 getContentFromUser (Todo)
- 2 getUsersWhoLikedContent (available thanks to @joelemanoel)
- 3 followUsers (available thanks to @mifi)

Please notice that if i use Follow Users Who Liked Content From This User

**Comment by RFlamant** [comment]
Indeed I'm talking here about an "enhancement" :) Thanks by advance 👍

**Comment by Flamintus** [comment]
@joelemanoel maybe you already have something like that!? :)

**Comment by joelemanoel** [comment]
Yeah. I will develop soon. Just find some free time and I'll do it.

**Comment by Flamintus** [comment]
Quick Share, for the moment, I run with that 👍 

'
    console.log('### PART 2 :STARTING following liker of my posts');
    console.log('Searching for liker of my posts');
    const myposts = ['CbuV2QhAnyN','CbwxgEyAcgc','CbtAG4jATiz','Cbs6nSTAIud','Cbs4pj8ADBb','Cbs3egsgh_K','Cbs2CV6ASfj','CbsgS_LAsHn','CbscPixgoZG','Cbr5JxtAM9'];
    for await (const post of myposts) {
        const getUsersWhoLikedContent= await instauto.getUsersWhoLikedContent({ contentId: post, maxPages: 10 }); console.log(getUsersWhoLikedContent);
        for await (const userstofollow of getUsersWhoLikedContent) {
            console.log('users to like ?', userstofollow)
            for await (const usertofollow of userstofollow) {
                console.log('usertolike', usertofollow)
                console.log('Following');
                await instauto.followUser(usertofollow);
            }
        }
    }
    console.log('DONE following liker of my posts');
'

Unfortunately, I dont have enough skill in NODEJS to : 
- dynamise post list form a username
- verify if we already follow a user before going on his profile

**Comment by metapodcod** [comment]
is there any progress on this?

**Comment by Flamintus** [comment]
I can share my actual peace of code that is working. BUT it is a bad way to do it I'm pretty sure, @joelemanoel may you help to integrate it properly ?

`    logger.info('####################################################');
    logger.info('### PART 1 :STARTING following liker of my posts ###');
    logger.info('####################################################');

    logger.info('### Searching for liker of my posts');

    const myposts = ['Ch3H6b0oolb', 'Ch1-dDVNf32', 'Chl59xhPyI-', 'Chh9CinNKoV'];
    logger.info('Post list : ', myposts);


    // END TODO

    // Lister mes Followers
    logger.info('### Searching for all previously folowed users');
    const allFollowing = await instauto.listManuallyFollowedUsers();
    logger.info('allFollowing', allFollowing);

    // On boucle sur les posts
    logger.info('### Starting Loop on my posts');
    for await (const post of myposts) {

        logger.info('####################################################');
        logger.info('### Searching for likers of postid ', post);
        logger.info('####################################################');
        const getUsersWhoLikedContent= await instauto.getUsersWhoLikedContent({ contentId: post, maxPages:10});  console.log(getUsersWhoLikedContent);

        // On boucle sur les flollowers du post
        for await (const userstofollow of getUsersWhoLikedContent) {

            logger.info('### Loop on Users', userstofollow)

            for await (const usertofollow of userstofollow) {

                await instauto.throttle();

                logger.info('####################################################');
                logger.info('### New USER', usertofollow);
                logger.info('####################################################');

                logger.info('[',usertofollow,']','Analysing');

                // EN COURS Si pas déja suivi
                logger.info('[',usertofollow,']','[CHECK][Recently Asked for follow] Runing');
                const allreadyaskedforfollow = await instauto.haveRecentlyFollowedUser(usertofollow);
                logger.info('[',usertofollow,']','[CHECK][Recently Asked for follow] Result', allreadyaskedforfollow);


                logger.info('[',usertofollow,']','[CHECK][Allready Folloing] Runing');
                const allreadyfollowed = allFollowing.find(allFollowing => {
                  return allFollowing === usertofollow
                })
                logger.info('[',usertofollow,']','[CHECK][Allready Folloing] Result', !!allreadyfollowed);

                if (allreadyfollowed) {
                    logger.info('[',usertofollow,']','----> SKIP (We are Allready Folloing this user)');
                }else if(allreadyaskedforfollow){
                    logger.info('[',usertofollow,']','----> SKIP (We Recently Asked this user for follow)');
                }else{
                    logger.info('[',usertofollow,']','[CONTINUE][Not folowing and not recently followed by me]');

                    // Todo Si non privé
                    logger.info('[',usertofollow,']','[CHECK][Does User Flow me] Runing');
                    const followsMe = await instauto.doesUserFollowMe(usertofollow);
                    logger.info('[',usertofollow,']','[CHECK][Does User Flow me] Result', followsMe);

                    if(followsMe != true){
                        logger.info('[FINISHED] Following user ', usertofollow);
                        logger.info('[',usertofollow,']','[FINISH][Following user] Runing');

                        await instauto.followUser(usertofollow);
                        logger.info('[FINISH][Following user] DONE', usertofollow);
                        logger.info('[',usertofollow,']','[FINISH][Following user] DONE');

                    }else{
                        logger.info('[',usertofollow,']','----> SKIP (We are allready followed by this user)');
                    }
                }
            }
        }
    }
    console.log('DONE following liker of my posts');`

**Comment by Flamintus** [comment]
Any news about this ?
